exchanges/poloniex/poloniex_types.go

```diff
@@ -446,75 +125,906 @@ var WithdrawalFees = map[currency.Code]float64{
+	Message       string `json:"message,omitempty"`
+}
+
+// CancelReplaceOrderRequest represents a cancellation and order replacement request parameter.
+type CancelReplaceOrderRequest struct {
+	orderID           string  `json:"-"` // orderID: used in order path parameter.
+	ClientOrderID     string  `json:"clientOrderId"`
+	Price             float64 `json:"price,omitempty,string"`
+	Quantity          float64 `json:"quantity,omitempty,string"`
+	Amount            float64 `json:"amount,omitempty,string"`
+	AmendedType       string  `json:"type,omitempty,string"`
+	TimeInForce       string  `json:"timeInForce,omitempty"`
+	AllowBorrow       bool    `json:"allowBorrow,omitempty"`
+	ProceedOnFailure  bool    `json:"proceedOnFailure,omitempty,string"`
+	SlippageTolerance float64 `json:"slippageTolerance,omitempty,string"`
+}
+
+// CancelReplaceOrderResponse represents a response parameter for order cancellation and replacement operation.
+type CancelReplaceOrderResponse struct {
+	ID            string       `json:"id"`
+	ClientOrderID string       `json:"clientOrderId"`
+	Price         types.Number `json:"price"`
+	Quantity      types.Number `json:"quantity"`
+	Code          int64        `json:"code"`
+	Message       string       `json:"message"`
+}
+
+// OrdersHistoryRequest holds a orders history request parameters
+type OrdersHistoryRequest struct {
+	Symbol      currency.Pair
+	AccountType string
+	OrderType   string
+	OrderTypes  []string
+	Side        string
+	Direction   string
+	States      string
+	From        int64
+	Limit       int64
+	StartTime   time.Time
+	EndTime     time.Time
+	HideCancel  bool
+}
+
+// TradeOrder represents a trade order instance.
+type TradeOrder struct {
+	ID             string            `json:"id"`
+	ClientOrderID  string            `json:"clientOrderId"`
+	Symbol         string            `json:"symbol"`
+	State          string            `json:"state"`
+	AccountType    string            `json:"accountType"`
+	Side           string            `json:"side"`
+	Type           string            `json:"type"`
+	TimeInForce    order.TimeInForce `json:"timeInForce"`
+	Quantity       types.Number      `json:"quantity"`
+	Price          types.Number      `json:"price"`
+	AveragePrice   types.Number      `json:"avgPrice"`
+	Amount         types.Number      `json:"amount"`
+	FilledQuantity types.Number      `json:"filledQuantity"`
+	FilledAmount   types.Number      `json:"filledAmount"`
+	CreateTime     types.Time        `json:"createTime"`
+	UpdateTime     types.Time        `json:"updateTime"`
+	OrderSource    string            `json:"orderSource"`
+	Loan           bool              `json:"loan"`
+	CancelReason   int64             `json:"cancelReason"`
+}
+
+// SmartOrder represents a smart order detail.
+type SmartOrder struct {
+	ID            string            `json:"id"`
+	ClientOrderID string            `json:"clientOrderId"`
+	Symbol        string            `json:"symbol"`
```

Comment:
🚧 Change request:
Unless I'm missing something:
```suggestion
	Symbol        currency.Pair            `json:"symbol"`
```